AuthenticationStrengthv1

InbarckMS opened the pull request:
Moving Auth Strength to v1 and depreacting functions from beta

jbzdarkid commented on api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md:
```diff
@@ -7,11 +7,15 @@ ms.prod: "identity-and-sign-in"
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
+[!CAUTION] 
+  > The `findByMethodMode` function is only available in `/beta` and will not become available in `/v1.0`. To get a list of  [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) filtered to only include policies that include the authentication method mode specified in the request, use XXX instead
```
```suggestion
  > The `findByMethodMode` function is only available in `/beta` and will not become available in `/v1.0`. To get a list of [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) filtered to only include policies that include the authentication method mode specified in the request, use XXX instead
```

FaithOmbongi commented on api-reference/v1.0/resources/authenticationstrengthpolicy.md:
```diff
@@ -0,0 +1,73 @@
+|[Delete authenticationStrengthPolicy](../api/authenticationstrengthroot-delete-policies.md)|None|Delete a custom [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object. You can't delete a built-in **authenticationStrengthPolicy** object.|
+|[List usage](../api/authenticationstrengthpolicy-usage.md)|[authenticationStrengthUsage](../resources/authenticationstrengthusage.md)|Find all [conditionalAccessPolicies](../resources/conditionalaccesspolicy.md) that reference an authentication strength.|
+|[updateAllowedCombinations](../api/authenticationstrengthpolicy-updateallowedcombinations.md)|[updateAllowedCombinationsResult](../resources/updateallowedcombinationsresult.md)|Update the allowed [authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md) for a given [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md).|
+|[List combinationConfigurations](../api/authenticationstrengthpolicy-list-combinationconfigurations.md)|[authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md) collection|Get the authenticationCombinationConfiguration resources from the combinationConfigurations navigation property.|
+|[Create authenticationCombinationConfiguration](../api/authenticationstrengthpolicy-post-combinationconfigurations.md)|[authenticationCombinationConfiguration](../resources/authenticationcombinationconfiguration.md)|Create a new authenticationCombinationConfiguration object.|
```
Not here. They should be in the authenticationCombinationConfiguration.md file.
```suggestion
```

InbarckMS replied:
I see them already in authenticationCombinationConfiguration, so I'm going to only remove them from here.

FaithOmbongi commented on api-reference/v1.0/resources/authenticationstrengthusage.md:
```diff
@@ -0,0 +1,49 @@
+---
```
This object isn't in the v1.0 schema. Please verify.
Once fixed, reassign the doc to yourself.

InbarckMS replied:
We need to add to the v1 PR, will keep this comment open until we address in the PR.

FaithOmbongi commented on api-reference/v1.0/api/authenticationstrengthpolicy-updateallowedcombinations.md:
```diff
@@ -0,0 +1,104 @@
+---
+title: "authenticationStrengthPolicy: updateAllowedCombinations"
+description: "Update the allowedCombinations collection for a given authentication strength policy object."
+author: "mmcla; inbarckMS"
```
Only one author.

FaithOmbongi commented on api-reference/v1.0/resources/authenticationcombinationconfiguration.md:
```diff
@@ -0,0 +1,57 @@
+
+Namespace: microsoft.graph
+
+An authentication combination configuration sets restrictions on specific types, modes, versions, etc. of an authentication method that is tied to specific auth method combinations used in an [authentication strength](authenticationstrengths-overview.md).
```
```suggestion
An authentication combination configuration sets restrictions on specific types, modes, and versions of an authentication method that is tied to specific auth method combinations used in an [authentication strength](authenticationstrengths-overview.md).
```

FaithOmbongi commented on api-reference/v1.0/resources/authenticationcombinationconfiguration.md:
```diff
@@ -0,0 +1,57 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|appliesToCombinations|authenticationMethodModes collection|Which authentication method combinations this configuration applies to. Must be an **allowedCombinations** object that's defined for the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md). The only possible value for fido2combinationConfigurations is `"fido2"`.|
+|id|String|A unique system-generated identifier. Inherited from [entity](../resources/entity.md).|
```
```suggestion
|id|String|A unique system-generated identifier.|
```

FaithOmbongi commented on api-reference/v1.0/resources/authenticationcombinationconfiguration.md:
```diff
@@ -0,0 +1,57 @@
+  "blockType": "resource",
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.authenticationCombinationConfiguration",
+  "baseType": "microsoft.graph.entity",
```
For same reason indicated above
```suggestion
```

FaithOmbongi commented on api-reference/v1.0/resources/authenticationmethodmodedetail.md:
```diff
@@ -0,0 +1,54 @@
+
+For more information on authentication methods, see the [authentication methods overview](../resources/authenticationmethods-overview.md)
+
+Inherits from [entity](../resources/entity.md).
```
Not anymore

FaithOmbongi commented on api-reference/v1.0/resources/authenticationmethodmodedetail.md:
```diff
@@ -0,0 +1,54 @@
+|:---|:---|:---|
+|authenticationMethod|baseAuthenticationMethod|The authentication method that this mode modifies. The possible values are: `password`, `voice`, `hardwareOath`, `softwareOath`, `sms`, `fido2`, `windowsHelloForBusiness`, `microsoftAuthenticator`, `temporaryAccessPass`, `email`, `x509Certificate`, `federation`, `unknownFutureValue`.|
+|displayName|String|The display name of this mode|
+|id|String|The system-generated identifier for this mode. Inherited from [entity](../resources/entity.md).|
```
Same reason as above
```suggestion
|id|String|The system-generated identifier for this mode. |
```

FaithOmbongi commented on api-reference/v1.0/resources/authenticationmethodmodedetail.md:
```diff
@@ -0,0 +1,54 @@
+  "blockType": "resource",
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.authenticationMethodModeDetail",
+  "baseType": "microsoft.graph.entity",
```
Same reason as above
```suggestion
```

FaithOmbongi commented on api-reference/v1.0/api/authenticationstrengthroot-list-authenticationmethodmodes.md:
```diff
@@ -0,0 +1,240 @@
+---
+title: "List authenticationMethodModes"
+description: "Get a list of all supported authentication methods, or all supported authentication method combinations."
+author: "mmcla;inbarckMS"
```
Only One

FaithOmbongi commented on api-reference/v1.0/resources/conditionalaccessgrantcontrols.md:
```diff
@@ -33,7 +33,10 @@ Consider the following when you use the `passwordChange` control:
-None.
+|Relationship|Type|Description|
+|:---|:---|:---|
+| authenticationStrength | [authenticationStrengthPolicy](authenticationstrengthpolicy.md) | The authentication strength required by the conditional access policy. Optional.|
```
This change isn't in the v1.0 schema PR. Please verify.

InbarckMS replied:
Auth strength is now updated in the CA API schema. Closing this comment.

FaithOmbongi reviewed (changes requested):
Thank you @InbarckMS for your work on this PR.

I've left suggestions inline plus the following:
- Question: Are updateAllowedCombinations action and findByMethodMode function getting deprecated in beta or is it that they won't be promoted to v1.0? 
   - If it's the latter, remove the "(deprecated)" statement from both the TOC and the titles.
   - If the former, also ensure that your schema PR includes the deprecation annotations. That will be required for approval of the docs PR.
- Add the **authenticationStrengthPolicies** relationship to [policyRoot](https://learn.microsoft.com/en-us/graph/api/resources/policyroot?view=graph-rest-1.0).
- GLOBAL: The enums like authenticationMethodModes, authenticationStrengthPolicyType, and authenticationStrengthRequirements haven't been defined in the enums.md file . This is the cause for many (if not all) of the API doctor errors in the v1.0 files.
- GLOBAL: I mentioned a while back that we need to start calling out the RBAC roles for delegated access READ vs WRITE. That way, customers don't have to use Global Admin, dig around, or report 403s. Let me know once you have the info available and I can guide you on this.
   - Let's do that for GA in this PR. You can raise a separate PR for beta.
- GLOBAL: Delete all the `# [HTTP](#tab/http)` in examples. These will be added by the snippets generator.
- Couldn't find neither the **usage** function nor the **authenticationStrengthUsage** complex type in the v1.0 schema PR. Please verify.
- Couldn't find the **authenticationStrength** relationship of conditionalAccessGrantControls in the schema either. Please verify.
- GLOBAL: I see some docs are still assigned to Michael. Please reassign them to yourself. Even the beta docs.
  - Also, only 1 alias is allowed in the "author" field.
- GLOBAL: None of the resources inherited from entity in v1.0. Please make sure you remove any "Inherits from entity/Inherited from entity/"baseType": "microsoft.graph.entity"" entries.
- GLOBAL: I see you moved all CA-related URLs from `/identity/conditionalAccess/authenticationStrengths/` (plural Strengths) to `/identity/conditionalAccess/authenticationStrengths/` (singular Strength). However, there are many left over instances even in beta.
   - Please do a search and replace to replace all these instances in both v1.0 and beta.
   - For response objects, the @odata.context also contains the plural form. Make sure to change to the singular form `...#identity/conditionalAccess/authenticationStrength/`.
   - This isn't a breaking change because you're not blocking the API side. But let's document the endpoint that you want customers to use going forward.
- Add the changelog please. Remember to download it from your schema PR in time. The artifacts expire after 15 days of a successful build and if the schema PR is merged and 15 days pass before you get the files, you will need to create the changelog manually.

InbarckMS commented on api-reference/beta/api/authenticationstrengthpolicy-updateallowedcombinations.md:
```diff
@@ -7,11 +7,12 @@ ms.prod: "identity-and-sign-in"
 ---
 
-# authenticationStrengthPolicy: updateAllowedCombinations
+# authenticationStrengthPolicy: updateAllowedCombinations 
```
We ended up not depreacting updateAllowedCombinations, only findByMethod.

InbarckMS commented on api-reference/v1.0/api/authenticationstrengthpolicy-usage.md:
```diff
@@ -0,0 +1,93 @@
+---
+title: "authenticationStrengthPolicy: usage"
+description: "The `usage` function allows the caller to see which Conditional Access policies reference a specified authentication strength policy"
```
Updated

InbarckMS commented on api-reference/v1.0/api/authenticationstrengthpolicy-usage.md:
```diff
@@ -0,0 +1,93 @@
+# authenticationStrengthPolicy: usage
+Namespace: microsoft.graph
+
+The `usage` function allows the caller to see which Conditional Access policies reference a specified authentication strength policy. The policies are returned in two collections, one containing Conditional Access policies that require an MFA claim and the other containing Conditional Access policies that do not require such a claim. Policies in the former category are restricted in what kinds of ...
```
Updated

InbarckMS commented on api-reference/v1.0/api/authenticationstrengthroot-list-policies.md:
```diff
@@ -0,0 +1,168 @@
+|Delegated (work or school account)|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+|Delegated (personal Microsoft account)|Not supported.|
+|Application|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+
```
These are the RBAC:

-	Conditional Access administrator (R/W)
-	Security Administrator (R/W)
-	Security Reader (Read only)

Can you share an example I can modify for auth strength?

InbarckMS replied:
RBAC is updated in beta and v1. Thanks for creating the includes files.

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengthpolicy.md:
```diff
@@ -0,0 +1,73 @@
+|[Get authenticationStrengthPolicy](../api/authenticationstrengthpolicy-get.md)|[authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md)|Read the properties and relationships of an [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object.|
+|[Update authenticationStrengthPolicy](../api/authenticationstrengthpolicy-update.md)|[authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md)|Update the properties of a custom [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object. You can't update a built-in **authenticationStrengthPolicy** object. |
+|[Delete authenticationStrengthPolicy](../api/authenticationstrengthroot-delete-policies.md)|None|Delete a custom [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) object. You can't delete a built-in **authenticationStrengthPolicy** object.|
+|[List usage](../api/authenticationstrengthpolicy-usage.md)|[authenticationStrengthUsage](../resources/authenticationstrengthusage.md)|Find all [conditionalAccessPolicies](../resources/conditionalaccesspolicy.md) that reference an authentication strength.|
```
We are going to add it to the PR, I'm keeping this discussion open until we do.

InbarckMS replied:
authenticationStrengthUsage is now added. Closing this comment.

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengthpolicy.md:
```diff
@@ -0,0 +1,73 @@
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|allowedCombinations|authenticationMethodModes collection|A collection of authentication method modes that are required be used to satify this authentication strength.|
```
Updated. Will check API doctor again after the next commit.

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengthpolicy.md:
```diff
@@ -0,0 +1,73 @@
+|createdDateTime|DateTimeOffset|The datetime when this policy was created.|
+|description|String|The human-readable description of this policy.|
+|displayName|String|The human-readable display name of this policy. <br><br>Supports `$filter` (`eq`, `ne`, `not` , and `in`). |
+|id|String|The system-generated identifier for this mode. Inherited from [entity](../resources/entity.md).|
```
Updated.

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengthroot.md:
```diff
@@ -0,0 +1,54 @@
+
+The **authenticationStrengthRoot** resource is the entry point for the authentication strengths object model.
+
+Inherits from [entity](../resources/entity.md).
```
Updated

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengthroot.md:
```diff
@@ -0,0 +1,54 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|authenticationCombinations|authenticationMethodModes collection|A collection of all valid authentication method combinations in the system.|
+|id|String|A system-generated identifier. Inherited from [entity](../resources/entity.md).|
```
Updated

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengthroot.md:
```diff
@@ -0,0 +1,54 @@
+  "blockType": "resource",
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.authenticationStrengthRoot",
+  "baseType": "microsoft.graph.entity",
```
Updated

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengths-overview.md:
```diff
@@ -0,0 +1,84 @@
+title: "Azure AD authentication strengths API overview"
+description: "Authentication strengths are sets of authentication method combinations that determine which methods your users can use to authenticate to access a resource."
+ms.localizationpriority: medium
+author: "mmcla"
```
Updated

InbarckMS commented on api-reference/v1.0/resources/authenticationstrengths-overview.md:
```diff
@@ -0,0 +1,84 @@
+author: "mmcla"
+ms.prod: "identity-and-sign-in"
+doc_type: "conceptualPageType"
+ms.date: 12/10/2022
```
Updated to today, but this is likely going to be published in April. Is that OK?

InbarckMS commented:
> @FaithOmbongi - thanks for the review. I've gone over the comment and global comment and addressed the feedback.

Here is additional info regarding your question:
Question: Are updateAllowedCombinations action and findByMethodMode function getting deprecated in beta or is it that they won't be promoted to v1.0?
Answer: We ended up only deprecating findByMethodMode; I've asked the engineers to update the PR. Will circle back once it is done,

Add the authenticationStrengthPolicies relationship to policyRoot >> Done.

GLOBAL: The enums haven't been defined in the enums.md file  >> Done.

GLOBAL: RBAC --> I've responded to your comment with the permissions. Can you share an example I should follow?

GLOBAL: Delete all the # [HTTP](#tab/http) in examples. >> Done.

Couldn't find neither the usage function nor the authenticationStrengthUsage complex type in the v1.0 schema PR
+ Couldn't find the authenticationStrength relationship of conditionalAccessGrantControls in the schema either. 
 >> These are missing from the PR, waiting for the engineers to update and will circle back.

Please verify.
GLOBAL: Author - Update the docs to my name.

GLOBAL: None of the resources inherited from entity in v1.0.  >> Done.

GLOBAL: Lefovers /identity/conditionalAccess/authenticationStrengths/  >> Done.

changelog >> Done.

Please review, I will also circle back once the missing portions from the schema PR are updated.

FaithOmbongi reviewed (changes requested):
Hi @InbarckMS - A few feedback items, mostly clean up; See inline +
- GLOBAL: Restore all snippet files to what they were. The snippets will be autogenerated because the tooling will detect that your examples have changed.
- GLOBAL: Confirm the URLs in all the v1.0 docs. I see URLs for beta endpoints in v1.0 docs.
- For the RBAC statement, I've pushed a commit to create the include file and add a sample to one doc. For the READ statement, check out authenticationstrengthroot-list-policies.md; For WRITE statement, check out authenticationstrengthroot-delete-policies.md.

InbarckMS commented on api-reference/v1.0/api/authenticationcombinationconfiguration-get.md:
````diff
@@ -0,0 +1,78 @@
+The following is an example of a request.
+
+``` http
+GET https://graph.microsoft.com/beta/identity/conditionalAccess/authenticationStrength/policies/0e371351-6419-4c8a-8047-61eef0212ffb/combinationConfigurations/133b68c4-503b-4e87-839a-6c286a27381b
````
Updated.

InbarckMS commented on api-reference/v1.0/api/authenticationcombinationconfiguration-update.md:
```diff
@@ -0,0 +1,81 @@
+
+## Response
+
+If successful, this method returns a `204 NO CONTENT` response code.
```
Updated.

InbarckMS commented on api-reference/v1.0/api/authenticationcombinationconfiguration-update.md:
````diff
@@ -0,0 +1,81 @@
+The following is an example of a request.
+
+``` http
+PATCH https://graph.microsoft.com/beta/identity/conditionalAccess/authenticationStrength/policies/0e371351-6419-4c8a-8047-61eef0212ffb/combinationConfigurations/4643f174-fe85-42b8-8b84-516775750a30
````
Updated.

InbarckMS commented on api-reference/v1.0/api/authenticationstrengthpolicy-list-combinationconfigurations.md:
```diff
@@ -0,0 +1,87 @@
+---
+title: "List combinationConfigurations"
+description: "Get the authenticationCombinationConfiguration objects for an authentication strength policy."
+author: "mmcla;inbarckMS"
```
Updated.

InbarckMS commented on api-reference/v1.0/api/authenticationstrengthpolicy-post-combinationconfigurations.md:
```diff
@@ -0,0 +1,100 @@
+---
+title: "Create authenticationCombinationConfiguration"
+description: "Create a new authenticationCombinationConfiguration object. In use, only fido2combinationConfigurations may be created, and these may only be created for custom authentication strength policies."
+author: "mmcla;inbarckMS"
```
Updated.

InbarckMS commented on api-reference/v1.0/api/authenticationstrengthpolicy-update.md:
```diff
@@ -0,0 +1,78 @@
+
+## Response
+
+If successful, this method returns a `204 NO CONTENT` response code.
```
Updated

InbarckMS commented on api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md:
```diff
@@ -20,9 +19,17 @@ One of the following permissions is required to call this API. To learn more, in
-|Application|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+Get a list of the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) objects and their properties filtered to only include policies that include the authentication method mode specified in the request.
+
+> [!NOTE]
+> The `findByMethodMode` function is only available in `/beta` and will not be available in `/v1.0`. Use the following syntax instead.
+>
+> `GET /policies/authenticationStrengthPolicies?$filter[allowedCombinations/any(x:x has 'sms, password')]`Get a list of the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) objects and their properties filtered to only include policies that include the authentication method mode specified in the request.
 
+> [!NOTE]
+> The `findByMethodMode` function is only available in `/beta` and will not be available in `/v1.0`. Use the following syntax instead.
+>
+> `GET /policies/authenticationStrengthPolicies?$filter[allowedCombinations/any(x:x has 'sms, password')]`
```
Updated

InbarckMS commented on api-reference/beta/api/authenticationstrengthpolicy-findbymethodmode.md:
```diff
@@ -20,9 +19,17 @@ One of the following permissions is required to call this API. To learn more, in
 |Delegated (work or school account)|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
-|Delegated (personal Microsoft account)|Not supported.|
-|Application|Policy.Read.All, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.AuthenticationMethod|
+Get a list of the [authenticationStrengthPolicy](../resources/authenticationstrengthpolicy.md) objects and their properties filtered to only include policies that include the authentication method mode specified in the request.
```
Updated

InbarckMS commented on api-reference/beta/api/authenticationmethodmodedetail-get.md:
Updated

InbarckMS commented on api-reference/beta/api/authenticationstrengthpolicy-list-combinationconfigurations.md:
We have end point for /combinationConfigurations (for example FIDO2 advanced settings) and for /combinations  (that replaces authenticationCombinations). This is combinationConfigurations

FaithOmbongi replied:
Ah understood. Thanks.

FaithOmbongi commented:
Hi @InbarckMS - Please use the latest changelog stubs. Because the engineer added a few more changes to his schema PR, there are some missing objects that should have been generated - including the deprecation of findByMethodMode for beta.

FaithOmbongi reviewed (approved):
Hi @InbarckMS - This is good to go when the schema is live.

I've retained the changes to the snippets. The team has confirmed that they'll overwrite and replace all of them but changing it in this PR has no effect.

FaithOmbongi commented:
PR is on hold as we wait for the API to be deployed.
